src/Functional.js

```diff
@@ -0,0 +1,43 @@
+import { useState } from 'react';
+
+const todoList = [
+    'Do React course',
+    'Go Shopping',
+    'Read',
+];
+
+function Functional() {
+    const [todo, setTodo] = useState(false);
+
+    const handleChange = (event) => {
+        setTodo(event.target.value);
+    } 
+
+    return (
+        <div style={{margin: '5px', padding: '5px'}}>
+            <h2>Todo List</h2>
+            {todoList.map((item, key) => {
+                return (
+                    <>
```

Comment:
1. We need to use keys in each map we write in react. The syntax:
For native element(`div` in this case)
```jsx
{items.map(item => (
  <div key={item.id}>...</div>
))}
```
For custom React Component:
```jsx
// MyCustomReactComponent.jsx
const MyCustomReactComponent = () => <div></div> // or whatever you want to do in the component
// Note that you don't accept `key` prop explicitly. You just need to pass `key` prop to the component and React will do all the things for you

// ParentOfMyCustomReactComponent.jsx
...
{items.map(item => (
  <MyCustomReactComponent key={item.id}>...</MyCustomReactComponent>
))}
```
For React.Fragment:
```jsx
{items.map(item => (
  <React.Fragment key={item.id}> {/* Keep in mind, we use <React.Fragment>...</React.Fragment>, not <>...</>. This is JSX API */}
     <div /> 
    <CustomReactComponent />
    {/* Whatsoever */}
  </React.Fragment>
))}
```
[Keys and List rendering Docs](https://beta.reactjs.org/learn/rendering-lists)

2. When we set keys in lists, it's better to use unique ids(from backend preferably or generate on FE with [crypto.randomUUID()](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID)). The reason: when we use indices as keys, if we insert new element in the beginning in the list with no keys(React uses indices as keys by default), react will rerender all the items each rerender because it will think all the list items change(key and component type change) 
[Article why it's better not to use indices as keys](https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/)

Reply:
Reading [beta.reactjs.org](https://beta.reactjs.org/) will really illuminate these gaps